src/Our.Umbraco.Ditto/Extensions/TypeInferenceExtensions.cs

```diff
@@ -90,5 +92,33 @@ public static IEnumerable<Type> GetBaseTypes(this Type type)
                 type = type.BaseType;
             }
         }
+
+        /// <summary>
+        /// Gets a all Type instances matching the specified class name with just non-namespace qualified class name.
+        /// </summary>
+        /// <param name="className">Name of the class sought.</param>
+        /// <param name="baseType"></param>
+        /// <returns>Types that have the class name specified. They may not be in the same namespace.</returns>
+        public static IEnumerable<Type> GetTypeByName(string className, Type baseType)
+        {
+            /* [ML] - Look away kids, forgive me. We could improve performance with an optional app setting to filter assemblies 
+             * but this method executes in low milliseconds, hopefully with caching of these results we'll be ok
+             */
+
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies().Where(i => !i.FullName.ToLower().StartsWith("system") && !i.FullName.ToLower().StartsWith("microsoft") && !i.FullName.ToLower().StartsWith("windows")).ToList();
```

Comment:
As we need to grab an instance of the the archetype model POCO(s) to populate, we can only get this via the archetype alias (string), this method checks the dlls classes in the application for types which inherit from the base Archetype Model and where the class name matches the alias. Thinking about it we may also need to check if the alias has been set via an attribute, but it could be easily extended for that. 

I have filtered out System.*, Microsoft.* and Windows.* dlls as I am pretty sure we would want to exclude them.

The results of this are cached and also in testing with a test install the execution time for sub 30ms, so seemed like a hit worth taking on first load of each type. 

Cheers

Sent from my iPhone

> On 8 Mar 2015, at 13:00, James South <notifications@github.com> wrote:
> 
> In src/Our.Umbraco.Ditto/Extensions/TypeInferenceExtensions.cs:
> 
> > @@ -90,5 +92,33 @@ public static IEnumerable<Type> GetBaseTypes(this Type type)
> >                  type = type.BaseType;
> >              }
> >          }
> > +
> > +        /// <summary>
> > +        /// Gets a all Type instances matching the specified class name with just non-namespace qualified class name.
> > +        /// </summary>
> > +        /// <param name="className">Name of the class sought.</param>
> > +        /// <param name="baseType"></param>
> > +        /// <returns>Types that have the class name specified. They may not be in the same namespace.</returns>
> > +        public static IEnumerable<Type> GetTypeByName(string className, Type baseType)
> > +        {
> > +            /* [ML] - Look away kids, forgive me. We could improve performance with an optional app setting to filter assemblies 
> > +             * but this method executes in low milliseconds, hopefully with caching of these results we'll be ok
> > +             */
> > +
> > +            var assemblies = AppDomain.CurrentDomain.GetAssemblies().Where(i => !i.FullName.ToLower().StartsWith("system") && !i.FullName.ToLower().StartsWith("microsoft") && !i.FullName.ToLower().StartsWith("windows")).ToList();
> What's this bit loading up?
> 
> —
> Reply to this email directly or view it on GitHub.
>